### PR TITLE
Backfill Results table

### LINKED ISSUE #3663: Backfill results table with existing results columns
### Background
To speed the migration from the existing results columns to the new results table, we're going to write out to both for a while. This will make handling things like the ReportStream upload and analytics dashboard a bit easier, as we won't have to write queries that read from both the table and column. This is a short term solution however - long term, we want to backfill the results table with the contents of the results columns.

### Action requested
In the production database, backfill the results table with the results columns. We should be able to read all the test event rows and for each, create an entry in the results table.

### Acceptance Criteria
Results table has been backfilled with all our test event results.

### Additional context
It may not be possible to do this migration in Liquibase, as Terraform restricts the time a given migration can run and this one will likely take a while. (See https://github.com/usds/prime-simplereport-docs/blob/main/incident_postmortem/01-20-2022-failed-migration.md for why this is problematic). We'll need to devote some thinking on this problem before starting the ticket.

### PR DESCRIPTION
**Pull request by emmastephenson**
# DATABASE PULL REQUEST

## Related Issue

Fixes #3663 

## Changes Proposed

- Backfills the Result table with the existing results stored in TestOrder

## Additional Information

I am not a SQL wizard by any means! If you see room for improvement, please let me know.

This query does roughly the following:
- Gets all TestOrders and joins to the Result table such that only TestOrders _without_ an entry in the Result table are selected.
- Also filters out TestOrders that don't yet have a result (so, orders that are cancelled without saving a result or are pending).
- Inserts a row into the Result table for each TestOrder that fits the above criteria. These rows have the same metadata (created_at, created_by, updated_at, updated_by) as the original TestOrder. 
- All Result rows are marked `is_deleted=false` because TestOrders don't have an `is_deleted` column to pull from. (It has order_status instead, but we don't actually care about the status for Results)
- All rows are backfilled as covid results.
- The result LOINCs are taken from [Translators.java](https://github.com/CDCgov/prime-simplereport/blob/b87b595768ec5e4ab8fcdb593e000b2ed39f44ff/backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java#L390-L390)

### Rollbacks
We have three options that I can see for rollbacks:
1. Drop results created before $DATE, where date is the date we rolled out the original changes (this option is gross)
2. Add another column to Result, `is_backfill`, that will be set to TRUE for rows added during the migration. For the rollback, drop all rows where `is_backfill` is set to true
3. Have a no-op rollback, so that Liquibase doesn't freak out but no changes are actually made. This isn't actually a schema change, just a migration, so in my opinion a rollback isn't strictly necessary. But happy to hear other opinions on this. 

### Deployment Plan
We'll manually run the SQL in each database before merging this code to `main`. This should prevent us from running into issues on prod. (The prod rollout will take a while given that we have 4.8M TestOrders that need to be backfilled, so this change will be done after-hours)

We can also use SQL `EXPLAIN` to get a rough idea of the time commitment before actually doing the backfill.

As discussed in the 5/12 engineering sync, ultimately we'd like to split database deployments to their own pipeline. I don't want to wait for that infrastructure change for this particular migration though, because this migration will allow us to start on some of the other multiplex work (see below).

### Next Steps
After this database change is in, it frees us up to read from the Result table in PXP, results filtering, and the dashboard. (#3664, #3755, #3756)

## Testing

- I tested this extensively locally. I'm also planning to run manually in a lower environment before rolling out more broadly.

## Checklist for Primary Reviewer

- [ ] Only database changes are included in this PR
- [ ] Any new tables or columns that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
- [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views (including re-granting permission to the no-PHI user if need be)
- [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] Rollback has been verifed locally and in a deployed environment

### CODE REVIEW

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -3951,4 +3951,31 @@ databaseChangeLog:
...
+                t.test_event_id,
+                t.internal_id,
+                (SELECT d.internal_id from ${database.defaultSchemaName}.supported_disease d WHERE d.name='COVID-19'),
+                CASE WHEN t.result='POSITIVE' THEN '260373001' WHEN t.result='NEGATIVE' THEN '260415000' WHEN t.result='UNDETERMINED' THEN '455371000124106' END,
+                t.result
+              FROM ${database.defaultSchemaName}.test_order t
+              LEFT JOIN ${database.defaultSchemaName}.result ON t.internal_id=${database.defaultSchemaName}.result.test_order_id 
+              WHERE ${database.defaultSchemaName}.result.test_order_id IS NULL AND t.result IS NOT NULL;
+      rollback:
+        sql: SELECT 'foo';
```
Comment (emmastephenson) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:3981]: Please see my note about rollbacks in the PR description - this is intended as a no-op rollback.

**Review by nickclyde — DISMISSED** [review]
Looks great, thanks for working on this Emma!

**Review by zdeveloper — COMMENTED** [review]
LGTM! tested it locally and it works as expected, thank you!

**Review by zdeveloper — DISMISSED** [review]

**Comment by rin-skylight** [comment]
From an ops perspective, I don't see anything wrong with this, so long as we're confident that we have DB snapshots ready to go in the event of a failure. Thank you for all the research into this!

I'll circle back and approve once the standing conflict is addressed. ^-^

**Comment by emmastephenson** [comment]
Re-requesting review as I've pulled in the latest merge to main.

**Review by rin-skylight — APPROVED** [review]
Thanks for this! Looking forward to watching it in action. 😁

**Review by nickclyde — APPROVED** [review]

**Pull request merged**